tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_printing_Printer.java

```diff
@@ -157,19 +161,34 @@ public void test_getPrinterList() {
 		/* If there aren't any printers, verify that the
 		 * printer list is empty.
 		 */
-		PrinterData list[] = Printer.getPrinterList();
-		if (list.length  == 1) {
-			if (SWT.getPlatform().equals("gtk")) {
-				/* Even though there is no default printer data,
-				 * on GTK it is still possible to have a print
-				 * to file backend
+		Stream<PrinterData> printerStream = Arrays.stream(Printer.getPrinterList());
+
+		if (SWT.getPlatform().equals("gtk")) {
+			/* Even though there is no default printer data,
+			 * on GTK it is still possible to have a print
+			 * to file backend
+			 */
+			printerStream = printerStream.filter(pd->{
+				if ("GtkPrintBackendFile".equals(pd.driver)) {
+					return false;
+				}
+				return true;
+			});
+		} else if (SWT.getPlatform().equals("win32")) {
+			printerStream = printerStream.filter(pd->{
+				/*
+				 * Windows also has some default (non hardware) printer by default
 				 */
-				assertEquals("GtkPrintBackendFile", list[0].driver);
-			}
-		} else {
-		assertEquals("printer list contains items even though there are no printers",
-				0 , list.length);
+				if ("Microsoft Print to PDF".equalsIgnoreCase(pd.name) || "Microsoft XPS Document Writer".equalsIgnoreCase(pd.name) || "Fax".equalsIgnoreCase(pd.name)) {
+					return false;
+				}
```

Comment:
IMO 'switch-case' would be bit more readable here, something like below:
switch (pd.name.toLowerCase()) {
case ("fax"):
case ("microsoft print to pdf"):
case ("microsoft xps document writer"):
	return false;
}

Reply:
done.